lib/routes.dart

```diff
@@ -76,8 +76,10 @@ class RouteConfiguration {
       if (regExpPattern.hasMatch(settings.name)) {
         final match = regExpPattern.firstMatch(settings.name);
         Map<String, String> groupNameToMatch = {};
-        for (String groupName in match.groupNames) {
-          groupNameToMatch[groupName] = match.namedGroup(groupName);
+        if (match.groupCount > 0) {
+          for (String groupName in match.groupNames) {
+            groupNameToMatch[groupName] = match.namedGroup(groupName);
+          }
```

Comment:
`groupNames` would return an empty iterator when `groupCount` is 0, so your change here should not affect the error you are seeing.

What version of Flutter are you using? And how do you reproduce the error that you get?

Reply:
Alright, I'll close this issue then.

It's beta branch based on this commit `2294d75bf`
You'll see the error immediately if you run the app. 

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v1.15.17, on Mac OS X 10.15.3 19D76, locale en-US)

[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.6)
[✓] Connected device (2 available)

• No issues found!
```

Reply:
Ah, I see. This app is built to be run on the master channel, so that is why you get this error. I will update the pubspec.yaml to avoid confusion for this.